### PR TITLE
[gha] clean up libra to diem rename in docker publish

### PR DESCRIPTION
**Pull request by sausagee**
## Motivation
Clean up the remnants in libra to diem renaming in docker publish. 

## Test Plan
CI

### CODE REVIEW

File: .github/workflows/daily.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
   audit:
     runs-on: ubuntu-latest
     container:
-      image: libra/build_environment:github-1
```
Comment (rexhoffman) [inline .github/workflows/daily.yml:14]: hmm this is a bit of a problem here.   github-1 tag is no longer used in diem, I think switching the tag to "master" would let this keep functioning.

**Review by rexhoffman — CHANGES_REQUESTED** [review]

File: .github/workflows/daily.yml (hunk 1 above)
Comment (sausagee) [reply .github/workflows/daily.yml:14]: Thanks for the heads up!  

Once #7871 landed, we would have the `main` image https://hub.docker.com/r/diem/build_environment/tags?page=1&ordering=last_updated.

Comment (sausagee) [reply .github/workflows/daily.yml:14]: Will update.

**Review by bmwill — APPROVED** [review]

**Pull request closed without merging**